Automattic/sensei
# Add new menu itema landing page for promo student groups feature

Fixes #5199

### Changes proposed in this Pull Request
- Add new menu item Groups to Free Sensei LMS menu
- Add badge with Pro new to the menu item
- Create a new marketing page that opens when selecting a groups menu item

### Note: this PR is dependent on sensei pro PR that will add a filter to DISABLE this page 

### Testing instructions
- For testing this feature you will need to toggle Sensei Pro (enable disable) OR have 2 wp instances running one with sensei pro installed and another without
- Make sure that dependent Sensei Pro PR is merged (if it isn't you will not be able to test the case where this page is NOT displayed)

**First case with Sensei Pro installed**
- You should see the Groups menu item
- It should not have a badge near its name
- When clicking on it it should open the Groups list table and not the new marketing page

**Second case with Sensei Pro NOT installed**
- You should see the Groups menu item
- It should have badge near it's name like in the picture: 
<img width="209" alt="image" src="https://user-images.githubusercontent.com/7208249/176967929-4d203cd0-6888-40fb-85e1-5bc66bff47d8.png">
- When you click on it you should see a new marketing page
<img width="1250" alt="image" src="https://user-images.githubusercontent.com/7208249/176967151-df6f83e2-0549-49fc-a4c3-833a4dd58b4d.png">

**Third step:** 
 Play around with marketing page on mobile, tablet, or smaller screen and desktop view. It should not break.

| Mobile | Tablet | Desktop |
| -- | -- | -- |
| <img width="459" alt="image" src="https://user-images.githubusercontent.com/7208249/176967289-191063a0-bde6-4d83-b10b-2cf235e0195c.png"> |  <img width="484" alt="image" src="https://user-images.githubusercontent.com/7208249/176967326-490f482d-e472-43cc-bd7f-7d506b164d47.png"> |  <img width="1256" alt="image" src="https://user-images.githubusercontent.com/7208249/176967344-f86db90e-5898-4a58-9513-3590112da8de.png"> |

**Fourth**
- The 2 buttons on the marketing page should lead to the sensei lms pricing page 
<img width="469" alt="image" src="https://user-images.githubusercontent.com/7208249/176967432-4bcfd767-6635-4dce-a236-c16255bd704a.png">


### New/Updated Hooks

Hook filter: **sensei_student_groups_hide**
Filters the Student groups promo landing page.
param: {bool} $sensei_student_groups_hide Whether to hide the Student Groups promo landing page.
return {bool} Whether to hide the Student groups landing page.
